test/blackbox-tests/test-cases/pkg/ocamlformat/binary-avoid-from-project-deps.t

```diff
@@ -0,0 +1,164 @@
+Make sure that $ dune fmt does not take the OCamlFormat binary inside the "dune-project"
+dependencies.
+
+  $ . ../helpers.sh
+  $ mkrepo
+
+Make a fake ocamlformat.0.26.2:
+  $ mkdir ocamlformat
+  $ cd ocamlformat
+  $ cat > dune-project <<EOF
+  > (lang dune 3.13)
+  > (package (name ocamlformat))
+  > EOF
+  $ cat > ocamlformat.ml <<EOF
+  > let version = "0.26.2"
+  > ;;
+  > let () = print_endline ("formatted with version "^version)
+  > EOF
+  $ cat > dune <<EOF
+  > (executable
+  >  (public_name ocamlformat))
+  > EOF
+  $ cd ..
+  $ tar -czf ocamlformat-0.26.2.tar.gz ocamlformat
+
+Make a fake ocamlformat.0.26.3:
+  $ cd ocamlformat
+  $ cat > ocamlformat.ml <<EOF
+  > let version = "0.26.3"
+  > let () = print_endline ("formatted with version "^version)
+  > EOF
+  $ cd ..
+  $ tar -czf ocamlformat-0.26.3.tar.gz ocamlformat
+  $ rm -rf ocamlformat
+
+Make a package for the fake OCamlformat library:
```

Comment:
I'm seeing a lot of inconsistencies in the way "ocamlformat" gets capitalized in this PR, between "ocamlformat", "OCamlformat" and "OCamlFormat". My preference would be to just use all lower-case "ocamlformat" since that's the name of the package and the executable, however as long as it's consistent I don't really mind. Again something that can be done in a later PR so as to not block the work that depends on this PR being merged.